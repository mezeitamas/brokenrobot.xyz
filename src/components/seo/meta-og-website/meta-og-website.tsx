import React from 'react';
import type { FunctionComponent } from 'react';

type MetaOgWebsiteProps = {
    siteName: string;
    title: string;
    description: string;
    url: string;
};

const MetaOgWebsite: FunctionComponent<MetaOgWebsiteProps> = ({ siteName, title, description, url }) => {
    return (
        <>
            <meta
                property="og:type"
                content="website"
            />

            <meta
                property="og:locale"
                content="en_US"
            />
            <meta
                property="og:site_name"
                content={siteName}
            />
            <meta
                property="og:title"
                content={title}
            />
            <meta
                property="og:description"
                content={description}
            />
            <meta
                property="og:url"
                content={url}
            />
        </>
    );
};

export { MetaOgWebsite };

export type { MetaOgWebsiteProps };
