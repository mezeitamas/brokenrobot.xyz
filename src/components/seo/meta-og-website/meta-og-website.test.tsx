import React from 'react';

import { create } from 'react-test-renderer';

import { MetaOgWebsite } from './meta-og-website';
import type { MetaOgWebsiteProps } from './meta-og-website';

describe('<MetaOgWebsite />', () => {
    test('should render', () => {
        // Given
        const props: MetaOgWebsiteProps = {
            siteName: 'Broken Robot',
            title: 'Blog',
            description: 'Blog posts',
            url: 'https://www.brokenrobot.xyz/blog/'
        };

        // When
        const component = create(<MetaOgWebsite {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
