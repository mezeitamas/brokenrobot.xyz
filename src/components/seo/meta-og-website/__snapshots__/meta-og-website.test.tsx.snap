// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaOgWebsite /> should render 1`] = `
[
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="en_US"
    property="og:locale"
  />,
  <meta
    content="Broken Robot"
    property="og:site_name"
  />,
  <meta
    content="Blog"
    property="og:title"
  />,
  <meta
    content="Blog posts"
    property="og:description"
  />,
  <meta
    content="https://www.brokenrobot.xyz/blog/"
    property="og:url"
  />,
]
`;
